sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/expressions/Literal.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2.expressions;
+
+import org.apache.spark.annotation.Experimental;
+import org.apache.spark.sql.types.DataType;
+
+/**
+ * Represents a constant literal value in the public expression API.
+ *
+ * @param <T> the Java type of a value held by the literal
```

Comment:
sorry that's not what i meant ... as a library developer, I want to know what values I should be expecting from Literal's type T. In Spark there are now two types of values that we expose publicly to users (used to be only one but you changed it to expose InternalRow). For example, for a date type, is T an int, or T java.util.Date?

Reply:
It should be int for a date. I'll update the doc.

Reply:
Did you add the doc somewhere?

Reply:
I'm still waiting on your response for moving the expressions package. When I hear back from you, I'll push all the changes.